File: atividade_nicolas/alunos/nicolas_g1/kleberson/duracao.py
momento_inicio = input('Hora inicial: ').split(':')
momento_final = input('Hora final: ').split(':')

hora_inicio = int(momento_inicio[0])
minuto_inicio = int(momento_inicio[1])

hora_final = int(momento_final[0])
minuto_final = int(momento_final[1])

horas = hora_final - hora_inicio
minutos = minuto_final - minuto_inicio

if hora_inicio >= 0 and hora_inicio <= 23:
    if minuto_inicio >= 0 and minuto_final <= 60:
        print('A duração é de {} hora(s) e de {} minutos'.format(horas, minutos))

File: atividade_nicolas/alunos/nicolas_g1/icaro/avaliacao1 - javascript/duracao.js
import { input, print } from "../io_utils.js"

function main() {

    const momento_inicio = (input("Digite uma hora de início para um intervalo (ex: 04:15): ").split(":").map(Number));
    const momento_final = (input("Digite uma hora de fim para um intervalo (ex: 04:15): ").split(":").map(Number));
    
    if(momento_inicio[0] > 23 || momento_inicio[0] < 0) {
        print("um dos valores digitados está inválido");
        main();
    } 
        
    if(momento_final[0] > 23 || momento_inicio[0] < 0) {
        print("um dos valores digitados está inválido");
        main();
    }

    if(momento_inicio[1] > 59 || momento_inicio[1] < 0) {
        print("um dos valores digitados está inválido");
        main();
    } 

    if(momento_final[1] > 59 || momento_inicio[1] < 0) {
        print("um dos valores digitados está inválido");
        main();
    }

    const [horas, minutos] = calcula_intervalo(momento_inicio, momento_final);
    print(`o tempo decorrido foi de ${horas} horas e ${minutos} minutos`);
    
}

function calcula_intervalo(momento_inicio, momento_final) {

    let horas = momento_final[0] - momento_inicio[0];
    let minutos;

    if(momento_final[1] >= momento_inicio[1]) minutos = momento_final[1] - momento_inicio[1];
    else if(momento_final[1] < momento_inicio[1]) {
        minutos = momento_final[1] + 60 - momento_inicio[1];
        horas--;
    }

    return [horas, minutos];

}

main();